sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/TestHelper.java

```diff
@@ -29,13 +27,11 @@
     static Stream<Arguments> getTestParameters() {
         // when this issues is closed, the newer version of junit will have better support for
         // cartesian product of arguments - https://github.com/junit-team/junit5/issues/1427
-        List<Arguments> argumentsList = new ArrayList<>();
-        getHttpClients()
-            .forEach(httpClient -> {
-                Arrays.stream(ConfigurationServiceVersion.values()).filter(TestHelper::shouldServiceVersionBeTested)
-                    .forEach(serviceVersion -> argumentsList.add(Arguments.of(httpClient, serviceVersion)));
-            });
-        return argumentsList.stream();
+        ConfigurationServiceVersion[] filteredKeyServiceVersion =
+            Arrays.stream(ConfigurationServiceVersion.values()).filter(TestHelper::shouldServiceVersionBeTested)
+                .toArray(ConfigurationServiceVersion[]::new);
+
+        return getArgumentsFromServiceVersion(Arrays.asList(filteredKeyServiceVersion));
```

Comment:
This method should know what strategy to apply to the service versions. We may need to run specific versions or all or rolling. So, just providing an array of service versions will not be sufficient. This method should also take in a strategy.